examples/data_preprocessing/police_killing/preprocessing_with_kb.ipynb

```diff
@@ -0,0 +1,1883 @@
+    "print(f\"Test data: {df_test.shape[0]}\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "In the develoment data, 4453 (21.534964696779188%) instances are positive and 16225 instances (78.46503530322082%) are negative.\n",
+      "In the test data, 8878 (18.40114411258731%) instances are positive and 39369 instances (81.5988558874127%) are negative.\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Positive and negative instances in dev and test data\n",
+    "positive_dev = df_dev.groupby(\"label\").count()[\"sample\"][1]\n",
+    "negative_dev = df_dev.groupby(\"label\").count()[\"sample\"][0]\n",
+    "positive_test = df_test.groupby(\"label\").count()[\"sample\"][1]\n",
+    "negative_test = df_test.groupby(\"label\").count()[\"sample\"][0]\n",
+    "print(f\"In the develoment data, {positive_dev} ({(100/df_dev.shape[0])*positive_dev}%) instances are positive and {negative_dev} instances ({(100/df_dev.shape[0])*negative_dev}%) are negative.\")\n",
+    "print(f\"In the test data, {positive_test} ({(100/df_test.shape[0])*positive_test}%) instances are positive and {negative_test} instances ({(100/df_test.shape[0])*negative_test}%) are negative.\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 1.5 Output classes\n",
+    "\n",
+    "Our task is to find out whether a sentence describes the killing of a person by the police or does not. That means, it is a binary classification task with two output classes. The number of classes is defined with the *num_classes* parameter."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "num_classes = 2"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 2 Get the Rules"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For this task, the rules will be the names of people found in both our data and the FE-Database. A rule matches if a name found in the FE-Database can also be found in a sample."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 2.1 Standardizing the Names"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The problem when trying to match the FE-Database with our samples is that there might be different version of single names. For their dataset, Keith. et al used a standardized name version and kept all alternative names in a list in the column *names_org*. If might be possible that the FE-Database contains name versions which do not match with the standardized name of Keith et al., but with one of the versions in the *names_org*-list. (For instance, someone is called *Steiney James Richards Jr.* in the FE-Database and *James Richard* in our samples.)\n",
```

Comment:
```suggestion
    "The problem when trying to match the FE-Database with our samples is that there might be different version of single names. For their dataset, Keith. et al used a standardized name version and kept all alternative names in a list in the column *names_org*. It might be possible that the FE-Database contains name versions, which do not match with the standardized name of Keith et al., but with one of the versions in the *names_org*-list. (For instance, someone is called *Steiney James Richards Jr.* in the FE-Database and *James Richard* in our samples.)\n",
```